broker.go

```diff
@@ -0,0 +1,124 @@
+package influxdb
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/influxdb/influxdb/messaging"
+)
+
+// Broker represents an InfluxDB specific messaging broker.
+type Broker struct {
+	*messaging.Broker
+
+	done   chan struct{}
+	server dataNodeStore
+
+	// send CQ processing requests to the same data node
+	currentCQProcessingNode *DataNode
+
+	// variables to control when to trigger processing and when to timeout
+	TriggerInterval     time.Duration
+	TriggerTimeout      time.Duration
+	TriggerFailurePause time.Duration
+}
+
+type dataNodeStore interface {
+	DataNodes() []*DataNode
+}
+
+// NewBroker returns a new instance of a Broker with default values.
+func NewBroker() *Broker {
+	b := &Broker{
+		TriggerInterval:     5 * time.Second,
+		TriggerTimeout:      20 * time.Second,
+		TriggerFailurePause: 1 * time.Second,
+	}
+	b.Broker = messaging.NewBroker()
+	return b
+}
+
+func (b *Broker) RunContinuousQueryLoop(d dataNodeStore) {
+	b.server = d
+	b.done = make(chan struct{})
+	go b.continuousQueryLoop()
```

Comment:
Pass in `b.done` to `continuousQueryLoop()` as an argument since you close and unset it in `Close()`. You can get in a situation where you're selecting from a `nil` `b.done`.